src/UsgsAstroProjectedLsSensorModel.cpp

```diff
@@ -0,0 +1,799 @@
+  m_projString = "";
+}
+
+//*****************************************************************************
+// UsgsAstroProjectedLsSensorModel Constructor
+//*****************************************************************************
+UsgsAstroProjectedLsSensorModel::UsgsAstroProjectedLsSensorModel() {
+  _no_adjustment.assign(UsgsAstroProjectedLsSensorModel::NUM_PARAMETERS, 0.0);
+}
+
+//*****************************************************************************
+// UsgsAstroProjectedLsSensorModel Destructor
+//*****************************************************************************
+UsgsAstroProjectedLsSensorModel::~UsgsAstroProjectedLsSensorModel() {
+  if (m_logger) {
+    m_logger->flush();
+  }
+}
+
+//---------------------------------------------------------------------------
+// Core Photogrammetry
+//---------------------------------------------------------------------------
+
+//***************************************************************************
+// UsgsAstroLsSensorModel::groundToImage
+//***************************************************************************
+csm::ImageCoord UsgsAstroProjectedLsSensorModel::groundToImage(
+    const csm::EcefCoord &ground_pt, double desired_precision,
+    double *achieved_precision, csm::WarningList *warnings) const {
+  
+  PJ_CONTEXT *C = proj_context_create();
+
+  /* Create a projection. */
+  PJ *isdProj = proj_create(C, (m_projString + " +type=crs").c_str());
+  if (0 == isdProj) {
+    MESSAGE_LOG(
+        spdlog::level::debug,
+        "Failed to create isd transformation object");
+    return csm::ImageCoord(0, 0);
+  }
+
+  /* Create the geocentric projection for our target */
+  std::string radius_a = "+a=" + std::to_string(m_majorAxis);
+  std::string radius_b = "+b=" + std::to_string(m_minorAxis);
+  std::string projString = "+proj=geocent " + radius_a + " " + radius_b + " +type=crs";
+  PJ *ecefProj = proj_create(C, projString.c_str());
+  if (0 == ecefProj) {
+    MESSAGE_LOG(
+        spdlog::level::debug,
+        "Failed to create geocent transformation object");
+    return csm::ImageCoord(0, 0);
+  }
+
+  // Compute the transformation from our ISIS projection to ecef
+  PJ *isdProj2ecefProj = proj_create_crs_to_crs_from_pj(C, isdProj, ecefProj, 0, 0);
+  PJ_COORD c_in;
+  c_in.xyz.x = ground_pt.x;
+  c_in.xyz.y = ground_pt.y;
+  c_in.xyz.z = ground_pt.z;
+  MESSAGE_LOG(
+      spdlog::level::info,
+      "Ground point {}, {}, {}",
+      c_in.xyz.x, c_in.xyz.y, c_in.xyz.z);
+  PJ_COORD c_out = proj_trans(isdProj2ecefProj, PJ_INV, c_in);
+  MESSAGE_LOG(
+      spdlog::level::info,
+      "Meters {}, {}",
+      c_out.xyz.x, c_out.xyz.y);
+  std::vector<double> lineSampleCoord = meterToPixel(c_out.xyz.x, c_out.xyz.y, m_geoTransform);
+  csm::ImageCoord imagePt(lineSampleCoord[0], lineSampleCoord[1]);
+  MESSAGE_LOG(
+      spdlog::level::info,
+      "groundToImage result of ({}, {})",
+      imagePt.line, imagePt.samp);
+  return imagePt;
```

Comment:
Seems like simply using proj was accurate enough for this without needing the camera model or image to ground?

Reply:
Yes(?) I don't think involving the camera model would make a difference. We are in ground, so we could go back to camera image, then back to ground but you would just get the same ground. So you just feed it through the projection to get back to the projected pixel